docs/relations.md

```diff
@@ -60,6 +60,48 @@ Also `has_many`, `belongs_to` and `has_one` relations have `dependent` parameter
 
 `has_many` and `has_one` relations also accepts `inverse_of` option which presents inverse relation name. Specifying this option will automatically load owner object during any relation loading (because of `ModelQuery#includes` or `ModelQuery#eager_load` or even plaint `SomeModel#relation_name` method call).
 
+## Examples of non-standard relationships
+
+Sometimes relationships between tables use non-standard attributes or may require more than a single field as keys for the relationship. In situations like this you can use the `request` parameter on the relationship. This parameter takes a block using the `where` statement syntax, and can even use the other model column as a reference.
+
+```crystal
+class Address < Jennifer::Model::Base
+  with_timestamps
+
+  belongs_to :person, foreign: :guid, primary: guid
+
+  # In this example model, the `type` field is a string of "mailing_address" or "physical_address"
+  mapping(
```

Comment:
This example came from how I was able to get Jennifer working with a DB that was already configured to work this way in a Rails app. It didn't properly use STI, but kind of mocked the behavior.

As far as where the `mapping` macro goes, I like to organize my models so the relationship definitions are at the top of the file, the `mapping` macro goes below that to make seeing the fields easy, and below that, I define any model methods. Just a personal preference because it makes finding relationships quick and easy.

Reply:
Currently all examples follow the opposite notation - mapping first and relations next. Could you please change the order here so it is consistent with other pages?